main.rb

```diff
@@ -0,0 +1,67 @@
+require 'open3'
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+ 
+def env_has_key(key)
+	value = get_env_variable(key)
+	return value unless value.nil? || value.empty?
+
+	abort("Input #{key} is missing.")
+end
+
+def run_command(command)
+	puts "@@[command] Executing: #{command}"
+	stdout_str, stderr_str, status = Open3.capture3(command)
+
+	if status.success?
+		puts stdout_str unless stdout_str.empty?
+		puts stderr_str unless stderr_str.empty?
+		return stderr_str.empty? ? stdout_str : stderr_str
+	else
+		puts "@@[error] Command failed:\n#{stderr_str}"
+		raise stderr_str
+	end
+end
+
+def abort_with1(message)
+	puts "@@[error] #{message}"
+	exit 1
+end
+
+def get_available_java_versions()
+	all_java_versions = [8, 11, 17, 21, 23]
```

Comment:
@csonuryilmaz  When I checked the documentation, I could not see the Java 23 version path. Is it pre-installed on runners?
cc:/ @ftmzhrsckldr 
https://docs.appcircle.io/infrastructure/android-build-infrastructure#java-version

Reply:
@ftmzhrsckldr  the java 23 is not available on runners 

https://capture.dropbox.com/pa4QFgXv91uJYrKk